### PR TITLE
Create table egg_machines_monsters

### PR DESCRIPTION
**Pull request by loopdeer**
SQL:
```
CREATE TABLE `egg_machines_monsters` (
  `egg_machine_monster_id` int(11) NOT NULL AUTO_INCREMENT,
  `monster_id` int(11) NOT NULL,
  `roll_chance` float NOT NULL,
  `egg_machine_id` int(11) NOT NULL,
  PRIMARY KEY (`egg_machine_monster_id`)
) ENGINE=InnoDB AUTO_INCREMENT=448 DEFAULT CHARSET=utf8mb4
```
egg_machines_monsters.py: Python SQL Item. egg_machine_monster_id is an auto-generated ID, monster_id is the monster's id, roll_chance is the chance to roll. egg_machine_id is the actual ID of the machine as according to it's ID in the egg_machine_table.

egg_machine_processor.py: Stores the monsters that appear in a new egg machine

egg_processor.py: New utility to read existing egg_machines (without their json files). Probably needs to be run the first time this is merged because older machines might not be in JSON.

### CODE REVIEW

**Comment by loopdeer** [comment]
Ready for review now, but let me know if I should add tstamp now. I have the commit ready to go for it!

**Comment by loopdeer** [comment]
Marked to draft to delete emm_id

**Comment by loopdeer** [comment]
Can't create multiple primary keys with the current SQL implementation in Python

File: etl/egg_processor.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        contents = ast.literal_eval(machine_sql['contents'])
+        for monster_id in contents.keys():
+            real_monster_id = int(monster_id.strip("()"))
+            emm = EggMachinesMonster(emm_id=None, monster_id=real_monster_id, roll_chance=contents.get(monster_id),
+                                     egg_machine_id=egg_machine_id)
+            db_wrapper.insert_or_update(emm)
+
+if __name__ == '__main__':
+    args = parse_args()
+    # This is a hack to make loading ES easier and more frequent.
```
Comment (chasehult) [inline etl/egg_processor.py:56]: These comments are left over from the last file, right.

File: etl/pad/storage/egg_machines_monsters.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "Monsters that appear in each egg_machine"
+    TABLE = 'egg_machines_monsters'
+    KEY_COL = 'emm_id'
+
+    def __init__(self,
+                 emm_id: int = None,
+                 monster_id: int = None,
+                 roll_chance: float = None,
+                 egg_machine_id: int = None):
+        self.emm_id = emm_id
```
Comment (chasehult) [inline etl/pad/storage/egg_machines_monsters.py:14]: Possibly rename to egg_machine_monster_id

**Review by chasehult — CHANGES_REQUESTED** [review]

File: etl/egg_processor.py (hunk 1 above)
Comment (loopdeer) [reply etl/egg_processor.py:56]: yeah...I will remove

File: etl/pad/storage/egg_machines_monsters.py (hunk 2 above)
Comment (loopdeer) [reply etl/pad/storage/egg_machines_monsters.py]: done, also edited the SQL code in the description

File: etl/egg_processor.py (hunk 1 above)
Comment (loopdeer) [reply etl/egg_processor.py]: Removed

**Pull request merged**